src/Features/text/TextUtils.tsx

```diff
@@ -0,0 +1,129 @@
+import { Group, Stack, Text, Textarea } from "@mantine/core";
+import { useDebouncedValue, useInputState } from "@mantine/hooks";
+
+const algorithms = ["gzip", "deflate", "zlib"];
+
+export default function TextUtils() {
+  const [text, setText] = useInputState("Sample text");
+
+  const getWordSet = () => {
+    const words = text
+      .replace(/[^a-zA-Z ]/g, "")
+      .split(" ")
+      .filter((word) => word.length > 0)
+      .map((word) => word.toLowerCase());
+    const wordSet = new Set(words);
+
+    const uniqueWords = wordSet.size;
+    const wordCount = words.length;
+
+    const density = uniqueWords / wordCount;
+
+    const stats = Array.from(wordSet).map((word) => ({
+      word,
+      count: words.filter((w) => w === word).length,
+    }));
+
+    return { stats, wordCount, uniqueWords, density };
+  };
+
+  const [retrieve] = useDebouncedValue(getWordSet, 500);
+  // Todo check why pasing retrieve directly to wordset throws an type error
+  const wordset = retrieve as unknown as {
```

Comment:
The issue here is `useDebounceValue` will debounce whatever you give it, you are giving it a function. 

so `retrieve` is a debounce function now. it's not type of `{ stats: {}, wordCount: ... }` etc. it's ` () => { stats: {}, wordCount: ... }`

useDebounceValue doesn't make sense here, you need `useMemo` with `text` as it's dependency, that should be enough. 
I don't know how it's working technically it shouldn't :P .  

ex:

```jsx
  const retrieve = useMemo(() => {
    const words = text
      .replace(/[^a-zA-Z ]/g, "")....

     // rest


    return { stats, wordCount, uniqueWords, density };
  }, [text]);

  const wordset = retrieve;
```

Reply:
let me know when you fix this, will merge after that

Reply:
Will do it later today 👍